ui/app/components/auth/form/okta.ts

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import AuthBase from './base';
+import { action } from '@ember/object';
+import { service } from '@ember/service';
+import { task, timeout } from 'ember-concurrency';
+import { tracked } from '@glimmer/tracking';
+import { waitFor } from '@ember/test-waiters';
+import errorMessage from 'vault/utils/error-message';
+import uuid from 'core/utils/uuid';
+
+import type AuthService from 'vault/vault/services/auth';
+
+/**
+ * @module Auth::Form::Okta
+ * see Auth::Base
+ * */
+
+export default class AuthFormOkta extends AuthBase {
+  loginFields = [{ name: 'username' }, { name: 'password' }];
+
+  @service declare readonly auth: AuthService;
+
+  @tracked challengeAnswer = '';
+  @tracked oktaVerifyError = '';
+  @tracked showNumberChallenge = false;
+
+  login = task(
+    waitFor(async (data) => {
+      // wait for 1s to wait to see if there is a login error before polling
+      await timeout(1000);
+
+      data.nonce = uuid();
+      this.pollForOktaNumberChallenge.perform(data.nonce, data.path);
+
+      try {
+        // selecting the correct okta verify answer on the personal device resolves this request
+        const authResponse = await this.auth.authenticate({
+          clusterId: this.args.cluster.id,
+          backend: this.args.authType,
+          data,
+          selectedAuth: this.args.authType,
+        });
+
+        this.onSuccess(authResponse);
+      } catch (error) {
+        // if a user fails the okta verify challenge, the POST login request fails (made by this.auth.authenticate above)
+        // bubble those up for consistency instead of managing error state in this component
+        this.onError(error as Error);
+        // cancel polling tasks and reset state
+        this.reset();
+      }
+    })
+  );
+
+  pollForOktaNumberChallenge = task(
+    waitFor(async (nonce, mountPath) => {
+      this.showNumberChallenge = true;
+
+      // keep polling /auth/okta/verify/:nonce API every 1s until response returns with correct_number
+      let verifyNumber = null;
+      while (verifyNumber === null) {
+        await timeout(1000);
+        // verifyNumber = await this.requestOktaVerify(nonce, mountPath);
+        verifyNumber = await this.requestOktaVerify(nonce, mountPath);
+      }
+
+      // display correct number so user can select on personal MFA device
+      this.challengeAnswer = verifyNumber ?? '';
+    })
+  );
```

Comment:
basically copy/pasted from:
https://github.com/hashicorp/vault/blob/741721ad0e2f1888114941a3009133cccdc4b7b2/ui/app/components/auth/login-form.js#L97-L116

Reply:
I removed the logic to return if `Ember.testing` because I opted to instead stub `uuid` and subsequently the okta/verify request using mirage